serverlessworkflow/modules/ROOT/pages/getting-started/create-your-first-workflow-service.adoc

```diff
@@ -1 +1,257 @@
+2022-05-25 14:38:13,380 INFO  [org.kie.kog.qua.pro.dev.DataIndexInMemoryContainer] (docker-java-stream--938264210) STDOUT: 2022-05-25 17:38:13,182 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, inmemory-postgres, jdbc-postgresql, narayana-jta, oidc, reactive-routes, rest-client-reactive, rest-client-reactive-jackson, security, smallrye-context-propagation, smallrye-graphql-client, smallrye-health, smallrye-metrics, smallrye-reactive-messaging, smallrye-reactive-messaging-http, vertx, vertx-graphql]
+----
+
+Once started, you can request the provided endpoint:
+
+[source,shell]
+----
+curl -X POST -H 'Content-Type:application/json' http://localhost:8080/hello_world
+----
+
+The response should be similar to:
+
+[source,shell]
+----
+{"id":"efb59bfa-ad9c-4062-a6d2-2d9184dd4b3d","workflowdata":{"greeting":"Hello World","mantra":"Serverless Workflow is awesome!"}}
+----
+
+Hit `CTRL+C` to stop the application, or keep it running and enjoy the blazing fast hot-reload.
+
+== Testing
+
+Now, let's add a test to make sure everything works as expected. But first, remove the
+auto-generated tests `org.kie.kogito.examples.GreetingResourceTest` and
+`org.kie.kogito.examples.NativeGreetingResourceIT`.
+
+Then, create the `org.kie.kogito.examples.HelloWorldTest` test class with the following content:
+
+[source,java]
+----
+package org.kie.kogito.examples;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+class HelloWorldTest {
+
+    @BeforeAll
+    static void init() {
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    }
+
+    @Test
+    void testHelloEndpoint() {
+        given()
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON)
+                .when().post("/hello_world")
+                .then()
+                .statusCode(201)
+                .body("workflowdata.greeting", is("Hello World"))
+                .body("workflowdata.mantra", is("Serverless Workflow is awesome!"));
+    }
+}
+----
+
+Now you're ready to run the test:
+
+[source,shell]
+----
+mvn clean verify
+----
+
+You can also run the test from your IDE directly (be sure you stopped the application first).
+
+If you want to know more about testing Quarkus applications, check out the https://quarkus.io/guides/getting-started-testing[Quarkus guide].
+
+== What's next?
+
+This guide covered the creation of a Serverless Workflow application using Kogito.
+However, there is much more. We recommend continuing the journey with the getting started guides
+link:cncf-serverless-workflow-specification-support.html[CNCF Serverless Workflow Specification Support]
```

Comment:
Good idea to add "Orchestration of OpenAPI based services".
Before removing "CNCF Serverless Workflow Specification Support", I just want to confirm, did you notice that this is not a link to the spec? It's a link to another guide under the getting-started category.

Reply:
I think that if a developer is looking for some feature / capability / use case they will first look at the available guides [1] and they should usually find the right guide there. Going into the details of the spec even if the CNCF guide / reference is an overview it is unnecessary and might be seen as too advanced content from my point of view. After the json in "=== Creating Your First Workflow" chapter it can be mentioned: "Did you know that the workflow definition is standard-based CNCF Serverless workflow?" and put there the link to guide for example.

[1] Understanding JQ expressions, Understanding workflow error handling, Handling events on workflows, etc.